### ISSUE TITLE
Compatibility Issue with Vite@5.2.3: TypeScript Type Mismatch Error

### ISSUE THREAD

**Issue by ldalves** [nkohari/apocrypha#2]
### Description

I'm encountering a TypeScript type mismatch error when trying to use `@nkohari/apocrypha` in a project that also uses Vite 5.2.3. It seems like the issue arises because `@nkohari/apocrypha` internally depends on Vite 4.1.1, leading to potential type and behavior discrepancies between the two versions.

### Steps to Reproduce

1. Set up a Vite project with Vite version 5.2.3.
2. Install `@nkohari/apocrypha` and configure it as described in the documentation.
3. Try building the project with `yarn build`.
4. Encounter the following TypeScript error:

```
TS2769: No overload matches this call. The last overload gave the following error. Type 'Plugin_2' is not assignable to type 'PluginOption'. ...
```

### Vite Config:

```
import path from 'path';
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react-swc';
import { apocrypha } from '@nkohari/apocrypha';
import tsconfigPaths from 'vite-tsconfig-paths';

// https://vitejs.dev/config/
export default defineConfig({
  build: {
    manifest: true,
  },
  plugins: [
    react(),
    tsconfigPaths(),
    apocrypha({
      paths: {
        components: 'src/components',
        content: 'content',
        declarations: 'src/markdoc',
      },
    }),
  ],
  server: { port: 3000 },
  resolve: {
    alias: {
      '@app': path.resolve(__dirname, 'src'),
    },
  },
});
```

### tsconfig:

```
{
  "compilerOptions": {
    "target": "ESNext",
    "useDefineForClassFields": true,
    "lib": ["DOM", "DOM.Iterable", "ESNext"],
    "module": "ESNext",
    "skipLibCheck": true,

    //    "moduleResolution": "bundler",
    "moduleResolution": "node",
    "allowImportingTsExtensions": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx",

    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true,

    "baseUrl": ".",
    "paths": { "@app/*": ["./src/*"] },

    "types": ["node", "vite/client"],
    "allowJs": false,
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": false,
    "forceConsistentCasingInFileNames": true
  },
  "include": ["src", "vite.config.ts"],
  "references": [{ "path": "./tsconfig.node.json" }]
}
```

### Expected Behavior
I would expect `@nkohari/apocrypha` to be compatible with the latest version of Vite or to gracefully handle version discrepancies to avoid type mismatches.

### Actual Behavior
TypeScript compilation fails due to a type mismatch error, which seems to stem from incompatible Vite versions between the project's direct dependency (`vite@5.2.3`) and `@nkohari/apocrypha`'s internal dependency (`vite@4.1.1`).

### Additional Context
Running `yarn why vite` in my project directory reveals the following dependency tree, showing the version mismatch:

- Found "vite@5.2.3" (project's direct dependency)
- Found "@nkohari/apocrypha#vite@4.1.1" (internal dependency of `@nkohari/apocrypha`)

This version mismatch may be the root cause of the type incompatibility issue. Is there a recommended approach to resolving this, or are there plans to update `@nkohari/apocrypha` to be compatible with Vite version 5.x?

Thank you for your great work! :)

**Comment by nkohari** [comment]
Sorry for the delay on this! I just published a new version of apocrypha (0.2.1) which moves to Vite 5.x. It also adds a couple of new features (static asset management and manifest management) which previously had to be implemented at the app level. I'm planning on adding some documentation for these features, but in the meantime the new version should fix the type discrepancy!

**Comment by ldalves** [comment]
Thank you very much for the update. I took a quick look at the changes and it's not very clear for now what the use of manifest management would be in this case. I saw that on your personal website you make use of a manifest provider. I understand that you are very busy developing your company and I look forward to an update to the documentation about these new features when you have time. :)
